x-pack/platform/plugins/shared/stack_connectors/public/connector_types/lib/servicenow/servicenow_connectors.tsx

```diff
@@ -41,20 +42,8 @@ const ServiceNowConnectorFields: React.FC<ActionConnectorFieldsProps> = ({
     notifications: { toasts },
   } = useKibana().services;
   const { updateFieldValues } = useFormContext();
-  const [{ id, isDeprecated, actionTypeId, name, config, secrets }] = useFormData<
-    ConnectorFormSchema<ServiceNowConfig, ServiceNowSecrets>
-  >({
-    watch: [
-      'id',
-      'isDeprecated',
-      'actionTypeId',
-      'name',
-      'config.apiUrl',
-      'config.isOAuth',
-      'secrets.username',
-      'secrets.password',
-    ],
-  });
+  const [{ id, isDeprecated, actionTypeId, name, config, secrets }] =
+    useFormData<ConnectorFormSchema<ServiceNowConfig, ServiceNowSecrets>>();
```

Comment:
This is the bug fix. Previously, we monitored only certain attributes of the config/secret, and the oauth fields were not considered.